qa/integration-tests/src/test/java/io/zeebe/broker/it/network/Ipv6IntegrationTest.java

```diff
@@ -0,0 +1,145 @@
+    configureGatewayContainer(gateway, initialContactPoints.get(0));
+  }
+
+  @After
+  public void tearDown() {
+    containers.parallelStream().forEach(GenericContainer::stop);
+  }
+
+  @Test
+  public void shouldCommunicateOverIpv6() {
+    // given
+    containers.parallelStream().forEach(GenericContainer::start);
+    gateway.start();
+
+    // when
+    final ZeebeClientBuilder zeebeClientBuilder =
+        ZeebeClient.newClientBuilder()
+            .usePlaintext()
+            .gatewayAddress(gateway.getExternalGatewayAddress());
+    try (final var client = zeebeClientBuilder.build()) {
+      final Topology topology = client.newTopologyRequest().send().join(5, TimeUnit.SECONDS);
+      // then - can find each other
+      TopologyAssert.assertThat(topology).isComplete(3, 1);
+    }
+  }
+
+  private void configureBrokerContainer(final int index, final List<ZeebeBrokerContainer> brokers) {
+    final int clusterSize = brokers.size();
+    final var broker = brokers.get(index);
+    final var hostNameWithoutBraces = getIpv6AddressForIndex(index);
+    final var hostName = String.format("[%s]", hostNameWithoutBraces);
+
+    initialContactPoints.add(hostName + ":" + ZeebePort.INTERNAL.getPort());
+
+    broker
+        .withNetwork(network)
+        .withNetworkAliases(NETWORK_ALIAS)
+        .withCreateContainerCmdModifier(
+            createContainerCmd ->
+                createContainerCmd
+                    .withIpv6Address(hostNameWithoutBraces)
+                    .withHostName(hostNameWithoutBraces))
+        .withEnv("ZEEBE_BROKER_NETWORK_MAXMESSAGESIZE", "128KB")
+        .withEnv("ZEEBE_BROKER_CLUSTER_NODEID", String.valueOf(index))
+        .withEnv("ZEEBE_BROKER_CLUSTER_CLUSTERSIZE", String.valueOf(clusterSize))
+        .withEnv("ZEEBE_BROKER_CLUSTER_REPLICATIONFACTOR", String.valueOf(REPLICATION_FACTOR))
+        .withEnv("ZEEBE_BROKER_CLUSTER_PARTITIONCOUNT", String.valueOf(PARTITION_COUNT))
+        .withEnv(
+            "ZEEBE_BROKER_CLUSTER_INITIALCONTACTPOINTS", String.join(",", initialContactPoints))
+        .withEnv("ZEEBE_BROKER_NETWORK_ADVERTISEDHOST", hostNameWithoutBraces)
+        .withEnv("ZEEBE_LOG_LEVEL", "DEBUG")
+        .withEnv("ZEEBE_BROKER_NETWORK_HOST", "::")
+        .withEnv("ZEEBE_LOG_LEVEL", "DEBUG")
+        .withEnv("ATOMIX_LOG_LEVEL", "INFO");
+  }
+
+  private void configureGatewayContainer(
+      final ZeebeGatewayContainer gateway, final String initialContactPoint) {
+    final String address = getIpv6AddressForIndex(CLUSTER_SIZE);
+    gateway
+        .withEnv("ZEEBE_GATEWAY_CLUSTER_CONTACTPOINT", initialContactPoint)
+        .withTopologyCheck(
+            new ZeebeTopologyWaitStrategy()
+                .forBrokersCount(CLUSTER_SIZE)
+                .forPartitionsCount(PARTITION_COUNT)
+                .forReplicationFactor(REPLICATION_FACTOR))
+        .withNetwork(network)
+        .withNetworkAliases(NETWORK_ALIAS)
+        .withEnv("ZEEBE_GATEWAY_NETWORK_HOST", "::")
```

Comment:
```suggestion
        .withEnv("ZEEBE_GATEWAY_NETWORK_HOST", "[::]")
        .withEnv("ZEEBE_GATEWAY_CLUSTER_HOST", address)
```

`ZEEBE_GATEWAY_CLUSTER_HOST` sets the address on which brokers should talk back to the gateway, so we also want it to be IPv6. Sorry I missed that initially.